test/unit/testplan/web_ui/test_web_app.py

```diff
@@ -0,0 +1,202 @@
+    """
+    Create the given file, and it's base directory if necessary.
+    Fill the file with the given contents.
+
+    :param tmp_file: Name of the file to be created (including path).
+    :type tmp_file: ``str``
+    :param contents: Contents of the file to be created.
+    :type contents: ``str``
+    """
+    base_dir = os.path.dirname(tmp_file)
+    if not os.path.exists(base_dir):
+        os.makedirs(base_dir)
+
+    with open(tmp_file, 'w') as tmp_file:
+        tmp_file.write(str(contents))
+
+
+def _create_tmp_static_files(base_dir):
+    """
+    Create fake static files to be sent by the Testplan web app.
+        <STATIC_PATH>/testplan/index.html
+        <STATIC_PATH>/testplan/static.file
+        <STATIC_PATH>/monitor/index.html
+        <STATIC_PATH>/monitor/static.file
+
+    :param base_dir: The Testplan web app's STATIC_PATH directory.
+    :type base_dir: ``str``
+    """
+    for report_type, report in list(STATIC_REPORTS.items()):
+        index_file = os.path.join(base_dir, report_type, 'index.html')
+        _create_tmp_file(tmp_file=index_file, contents=report['contents'])
+        static_file = os.path.join(base_dir, report_type, 'static.file')
+        _create_tmp_file(tmp_file=static_file, contents=report['contents'])
+
+
+def _create_tmp_data_files(base_dir):
+    """
+    Create fake data files to be sent by the Testplan web app.
+        <DATA_PATH>/reports/testplan_<UID>/report.json
+        <DATA_PATH>/reports/monitor_<UID>/report.json
+
+    :param base_dir: The Testplan web app's DATA_PATH directory.
+    :type base_dir: ``str``
+    """
+    for report_type, report in list(DATA_REPORTS.items()):
+        report_file = os.path.join(base_dir,
+                                   'reports',
+                                   '{}_{}'.format(report_type, report['uid']),
+                                   'report.json')
+        _create_tmp_file(tmp_file=report_file, contents=report['contents'])
+
+
+class TestStaticEndpoints(object):
+    """
+    Test the endpoints returning files from the STATIC_PATH directory.
+    """
+
+    def setup_method(self, _):
+        """Create fake static files and a test client."""
+        self.static_dir = tempfile.mkdtemp()
+        _create_tmp_static_files(self.static_dir)
+        tp_web_app.config['STATIC_PATH'] = self.static_dir
+        tp_web_app.config['TESTING'] = True
+        self.client = tp_web_app.test_client()
+
+    def teardown_method(self, _):
+        """Remove fake static files."""
+        shutil.rmtree(self.static_dir)
+
+    def test_testplan_index(self):
+        """Does /testplan/<uid> return the correct index.html page."""
+        # Use correct UID, expect 200 response.
+        response = self.client.get('/testplan/{}'.format(STATIC_REPORTS['testing']['uid']))
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['testing']['contents']) in str(response.data)
+
+        # Use incorrect UID, still expect 200 response.
```

Comment:
Do we not want a 404 response for an incorrect UID?

Reply:
Thanks I'll change it

Reply:
No, this endpoint is a little confusing. The UID isn't used by this endpoint, it is actually used by the index.html page returned by the endpoint. The index.html page gets the UID from the URL and makes the /testplan/<uid>/report request based on this. This is confusing, I would prefer it if the flask web app could inject this UID to the index.html page. However I couldn't get that to work and have the index.html page pass this UID to the react component when I tried. If you'd like to give it a go or have another idea to make this less confusing/more let me know :)